utils/dom.js

```diff
@@ -133,11 +129,29 @@ export function isVerticalEdge( container, isReverse, collapseRanges = false ) {
 	return true;
 }
 
+/**
+ * Get the rectangle of a given Range.
+ *
+ * @param {Range} range The range.
+ *
+ * @return {DOMRect} The rectangle.
+ */
+export function getClientRect( range ) {
+	// Adjust for empty containers.
+	return range.startContainer.nodeType === ELEMENT_NODE ?
+		range.startContainer.getBoundingClientRect() :
+		range.getClientRects()[ 0 ];
+}
+
+/**
+ * Get the rectangle for the selection in a container.
+ *
+ * @param {Element} container Editable container.
+ *
+ * @return {?DOMRect} The rectangle.
+ */
 export function computeCaretRect( container ) {
-	if (
-		includes( [ 'INPUT', 'TEXTAREA' ], container.tagName ) ||
```

Comment:
Why this removal? I'm guessing based on the remainder of the function that this might not have behaved well with non-contenteditables... is that correct?

Reply:
Yeah so I’m no sure why this is here... this function is specific to content editable... I can’t think of a reason to keep that here?